client/blocks/comment-detail/comment-detail-author.jsx

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import Gridicon from 'gridicons';
+import { localize } from 'i18n-calypso';
+import classNames from 'classnames';
+
+export class CommentDetailAuthor extends Component {
+	static propTypes = {
+		author: PropTypes.object,
+		commentDate: PropTypes.string,
+	};
+
+	state = {
+		isExpanded: false,
+	};
+
+	toggleExpanded = () => {
+		this.setState( { isExpanded: ! this.state.isExpanded } );
+	}
+
+	render() {
+		const {
+			author,
+			commentDate,
+			moment,
+			translate,
+		} = this.props;
+		const { isExpanded } = this.state;
+
+		const classes = classNames( 'comment-detail__author', {
+			'is-expanded': isExpanded,
+		} );
+
+		return (
+			<div className={ classes }>
+				<div className="comment-detail__author-preview">
+					<div className="comment-detail__author-avatar">
+						<img src={ author.avatar_URL } />
```

Comment:
Oh, about this: I think the moment I started dropping the flat props in favour of nested objects was actually when I've introduced `ReaderAvatar`, that has an `author` prop instead of 4 different flat props.
See the mock author in its doc file: https://github.com/Automattic/wp-calypso/blob/master/client/blocks/reader-avatar/docs/example.jsx#L12-L17

I've taken the same mock author, and in no time I've found myself nesting everything into everything.